src/runtime/pay-flow.js

```diff
@@ -138,7 +138,7 @@ export class PayStartFlow {
         'swg': swgPaymentRequest,
         'i': {
           'startTimeMs': Date.now(),
-          'googleTransactionId': this.analyticsService_.getTransactionId(),
+          'googleTransactionId': this.payClient_.googleTransactionId_,
```

Comment:
Question: Are you absolutely certain that all paths set payClient_.googleTransactionId_?  It seemed like there were 2 major paths and one of them had fewer parameters.  I didn't trace the code closely enough to answer this question.  The existing code should always have the correct value so it seems safer to leave it as is.  Certainly arguable either way though.

Reply:
Here the googleTransactionId is set in i param as a part of payment request, which will be passed to PaymentsAsyncClient.loadPaymentData. Before this step, PaymentsAsyncClient is initialized with another set of option, where I pass the googleTransactionId because the constructor in gpay_async.js checks if googleTransactionId is set in i param. If not, a new id will be generated by gpay. In summary the previous setup does not pass transaction id until loadPaymentData, but it was checked by gpay in the constructor. Hope this makes sense.

Reply:
I was curious about whether payClient.googleTransactionId_ would get set for whatever path is using PayClientBindingSwg.

Reply:
It is not being used in prod because flag GPAY_API is set to be 100% in prod so I did not debug that path.

Reply:
Sorry if this line of questioning is annoying, but is that true in AMP also?

Reply:
I think so. Also SwG in AMP does not use gpay_async.js for the payment.